### DISCUSSION TITLE
kdb.DPolygon

### DISCUSSION THREAD

**Discussion by SMSajadi99** [gdsfactory/gdsfactory#3037] [Q&A] [answered]
Hello
I had a question for you, I just came across your nice project.
I did not understand a part of your tutorial:

```
gf.kdb.DPolygon
```
What does kdb mean and what does it do?

**Comment by joamatab** [comment]
Klayout DataBase

**Reply by aroop25** [reply]
HI,

Whenever I try importing kdb as  p1 = gf.kdb.DPolygon([(-8, -6), (6, 8), (7, 17), (9, 5)]), I get this error:
AttributeError: module 'gdsfactory' has no attribute 'kdb'

Can you tell me how to address this?

**Reply by joamatab** [reply]
Pip install gdsfactory--upgrade 


To probably have an old version of gdsfactory

**Reply by aroop25** [reply, marked as answer]
That worked.. Thanks

**Comment by SMSajadi99** [comment]
@joamatab thanks for attention, I had another question for you

```
add_polygon([(-8, -6), (6, 8), (7, 17), (9, 5)], layer=(1, 0))
```
and

```
gf.kdb.DPolygon
```

What is the difference between the above two methods (I feel they work the same, right?)

**Comment by joamatab** [comment]
yes, the first one ends up calling the second

**Comment by SMSajadi99** [comment]
![image](https://github.com/user-attachments/assets/b59ec877-4d64-4215-9d7d-e1d104fe097b)

@joamatab  I have a question for you:
How to increase the screen's height so that I can see all the shapes; Because a part of the shape is cut.

**Reply by sebastian-goeldi** [reply]
I am not sure I fully follow what you mean by screen height. I think that's rather a VSCode or window manager (or maybe browser) issue?
